src/dotnet/Notification.Service/NewChatEntryNotificationGenerator.cs

```diff
@@ -0,0 +1,49 @@
+using ActualChat.Chat;
+using ActualChat.Chat.Events;
+
+namespace ActualChat.Notification;
+
+public class NewChatEntryNotificationGenerator: IChatEventNotificationGenerator<NewChatEntryEvent>
+{
+    private readonly IChatAuthorsBackend _chatAuthorsBackend;
+    private readonly IChatsBackend _chatsBackend;
+    private readonly MomentClockSet _clocks;
+
+    public NewChatEntryNotificationGenerator(
+        IChatAuthorsBackend chatAuthorsBackend,
+        IChatsBackend chatsBackend,
+        MomentClockSet clocks)
+    {
+        _chatAuthorsBackend = chatAuthorsBackend;
+        _chatsBackend = chatsBackend;
+        _clocks = clocks;
+    }
+
+    // TODO(AK): support mentions
+    // TODO(AK): throttle notifications
+    public async IAsyncEnumerable<NotificationEntry> GenerateNotifications(
+        NewChatEntryEvent chatEvent,
+        [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        var clock = _clocks.CoarseSystemClock;
+        var title = await GetTitle(chatEvent.ChatId, cancellationToken).ConfigureAwait(false);
+        var content = GetContent(chatEvent.Content);
+            yield return new ChatNotificationEntry(chatEvent.ChatId, chatEvent.AuthorId, title, content, clock.Now);
```

Comment:
not clear why indent is here. Isn't there any formatter being applied automatically?